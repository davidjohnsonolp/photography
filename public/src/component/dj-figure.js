import buildShadowRoot from '../lib/build-shadow-root.js';

class DJFigure extends HTMLElement {
  constructor() {
    super();
    const html = `
      <style>
        :host {
        }
        figure {
          margin: 0 10px;
        }
        figure img {
          width: 100%;
        }
        figure figcaption {
          color: dimgray;
          font-style: italic;
          text-align: center;
        }
      </style>
      <figure>
        <img />
        <figcaption>
          <slot></slot>
        </figcaption>
      </figure>
    `;
    buildShadowRoot(html, this);

    this._img = this.shadowRoot.querySelector('img');
  }

  static get observedAttributes() {
    return ['src', 'alt'];
  }

  attributeChangedCallback(attrName, oldVal, newVal) {
    switch(attrName){
      case 'src':
        this._img.setAttribute('src', newVal);
        break;
      case 'alt':
        this._img.setAttribute('alt', newVal);
        break;
      default:
        break;
    }
  }
}

customElements.define('dj-figure', DJFigure);
export default DJFigure;
