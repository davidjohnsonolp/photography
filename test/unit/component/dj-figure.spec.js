import '../../../public/src/component/dj-figure.js';

let customFigure;

describe('DJFigure Custom Figure', () => {
  beforeEach(() => {
    customFigure = window.document.createElement('dj-figure'); 
    document.body.appendChild(customFigure);
  });

  afterEach(() => {
    document.body.getElementsByTagName('dj-figure')[0].remove();
  });

  it('renders correctly', () => {
    expect({ html: customFigure.shadowRoot.innerHTML }).toMatchSnapshot();
  });

  it('renders correctly with src', () => {
    customFigure.setAttribute('src', 'http://www.test.com/test.jpg');

    expect({ html: customFigure.shadowRoot.innerHTML }).toMatchSnapshot();
  });

  it('renders correctly with alt', () => {
    customFigure.setAttribute('alt', 'Test Alt');

    expect({ html: customFigure.shadowRoot.innerHTML }).toMatchSnapshot();
  });
});
