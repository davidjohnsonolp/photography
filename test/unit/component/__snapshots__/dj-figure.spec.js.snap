// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DJFigure Custom Figure renders correctly 1`] = `
Object {
  "html": "
      <style>
        :host {
        }
        figure {
          margin: 0 10px;
        }
        figure img {
          width: 100%;
        }
        figure figcaption {
          color: dimgray;
          font-style: italic;
          text-align: center;
        }
      </style>
      <figure>
        <img>
        <figcaption>
          <slot></slot>
        </figcaption>
      </figure>
    ",
}
`;

exports[`DJFigure Custom Figure renders correctly with alt 1`] = `
Object {
  "html": "
      <style>
        :host {
        }
        figure {
          margin: 0 10px;
        }
        figure img {
          width: 100%;
        }
        figure figcaption {
          color: dimgray;
          font-style: italic;
          text-align: center;
        }
      </style>
      <figure>
        <img alt=\\"Test Alt\\">
        <figcaption>
          <slot></slot>
        </figcaption>
      </figure>
    ",
}
`;

exports[`DJFigure Custom Figure renders correctly with src 1`] = `
Object {
  "html": "
      <style>
        :host {
        }
        figure {
          margin: 0 10px;
        }
        figure img {
          width: 100%;
        }
        figure figcaption {
          color: dimgray;
          font-style: italic;
          text-align: center;
        }
      </style>
      <figure>
        <img src=\\"http://www.test.com/test.jpg\\">
        <figcaption>
          <slot></slot>
        </figcaption>
      </figure>
    ",
}
`;
